### PR TITLE
Kr/big refactor

### PR DESCRIPTION
**Pull request by rocktavious**
Refactor the whole of the library so that its more maintainable but continues to work the same

## Changelog

- [x] List your changes here
- [x] Make a `changie` entry

## Tophatting

As seen in demo the commands `service preview` , `service import` and `service reconcile` still work the same.  Additionall i've also updated `service collect` which is lesser known and used but it works.

### CODE REVIEW

**Review by davidbloss — DISMISSED** [review]
LGTM! 🚀

File: src/cmd/config.go [hunk 1]
```diff
@@ -130,41 +49,37 @@ var configSampleCmd = &cobra.Command{
...
 	rootCmd.AddCommand(configCmd)
-	configCmd.AddCommand(
-		configSchemaCmd,
-		configViewCmd,
-		configSampleCmd,
-	)
+	configCmd.AddCommand(configSchemaCmd, configViewCmd, configSampleCmd)
 
-	configSampleCmd.Flags().Bool("simple", false, "Adjust the sample config to a bit simpler")
+	configSampleCmd.Flags().Bool("simple", false, "Adjust the sample config to be a less complex")
```
Comment (davidbloss) [inline src/cmd/config.go:68]: 🧹  `to be less complex` - please drop the `a`

File: src/cmd/preview.go [hunk 2]
```diff
@@ -3,108 +3,69 @@ package cmd
...
-	output := make([]common.ServiceRegistration, samples)
-	rand.Seed(time.Now().UTC().UnixNano())
-	for i, index := range getSamples(0, max, samples) {
-		output[i] = data[index]
-	}
-	return output
-}
+	prettyJSON, err := json.MarshalIndent(sampled, "", "    ")
+	cobra.CheckErr(err)
+	fmt.Printf("%s\n", string(prettyJSON))
```
Comment (davidbloss) [inline src/cmd/preview.go:59]: Fine as is but could also do `fmt.Println(string(prettyJSON))`

File: src/cmd/root.go [hunk 3]
```diff
@@ -59,15 +59,13 @@ func init() {
...
-	cobra.OnInitialize(initConfig)
-}
-
-func initConfig() {
-	readConfig()
-	setupLogging()
-	setupOutput()
-	setupConcurrency()
-	setupAPIToken()
+	cobra.OnInitialize(func() {
```
Comment (davidbloss) [inline src/cmd/root.go:62]: Nice! ✨

**Review by davidbloss — APPROVED** [review]
One grammar comment but overall LGTM 🎉 🚀

File: src/common/reconciler_test.go [hunk 4]
```diff
@@ -0,0 +1,366 @@
...
+				Name:     "Test",
+				Language: "Python",
+				Tier:     "tier_1",
+			},
+			service: service2,
+			result:  false,
+		},
+	}
+	// Act
+	autopilot.RunTableTests(t, cases, func(t *testing.T, test TestCase) {
```
Comment (taimoor-at-opslevel) [inline src/common/reconciler_test.go:362]: Cool!

File: src/common/reconciler_test.go [hunk 5]
```diff
@@ -0,0 +1,366 @@
...
+import (
+	"fmt"
+	"github.com/opslevel/kubectl-opslevel/common"
+	"github.com/opslevel/opslevel-go/v2023"
+	opslevel_jq_parser "github.com/opslevel/opslevel-jq-parser/v2023"
+	"github.com/rocktavious/autopilot/v2023"
+	"testing"
+)
+
+func Test_Reconciler_Reconcile(t *testing.T) {
```
Comment (taimoor-at-opslevel) [inline src/common/reconciler_test.go:12]: Non-blocking

I think it's unnecessary to have underscores here to make the function's name and purpose more clear. The tests are in a file dedicated to Reconciler testing, so having it all as one word `TestReconcilerReconcile` is readable enough.

**Review by taimoor-at-opslevel — APPROVED** [review]

File: src/cmd/preview.go (hunk 2 above)
Comment (rocktavious) [inline src/cmd/preview.go:59]:
```suggestion
	fmt.Println(string(prettyJSON))
```

File: src/cmd/config.go (hunk 1 above)
Comment (rocktavious) [inline src/cmd/config.go:68]:
```suggestion
	configSampleCmd.Flags().Bool("simple", false, "Adjust the sample config to be less complex")
```

**Pull request merged**